contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -109,13 +109,122 @@ namespace eosiosystem {
       _gstate2.new_ram_per_block = bytes_per_block;
    }
 
-   void system_contract::setparams( const eosio::blockchain_parameters& params ) {
+#ifdef EOSIO_SYSTEM_BLOCKCHAIN_PARAMETERS
+   extern "C" [[eosio::wasm_import]] void set_parameters_packed(const void*, size_t);
+#endif
+
+   void system_contract::setparams( const blockchain_parameters_t& params ) {
       require_auth( get_self() );
       (eosio::blockchain_parameters&)(_gstate) = params;
       check( 3 <= _gstate.max_authority_depth, "max_authority_depth should be at least 3" );
+#ifndef EOSIO_SYSTEM_BLOCKCHAIN_PARAMETERS
       set_blockchain_parameters( params );
+#else
+      constexpr size_t param_count = 18;
+      // an upper bound on the serialized size
+      char buf[sizeof(params) + param_count];
```

Comment:
+ 1

Reply:
It'll actually work, because `binary_extension` has an extra bool that isn't serialized.